swarmcg/config.py

```diff
@@ -38,42 +30,43 @@
-fct_guess_min_flat_diff_angles = 50 # flat minimum force constant variation that fct_guess_fact shall yield, used to find low and high boundaries for random generation of particles' force constants
-fct_guess_min_flat_diff_dihedrals_without_mult = 0.50 # flat minimum force constant variation that fct_guess_fact shall yield, used to find low and high boundaries for random generation of particles' force constants
-fct_guess_min_flat_diff_dihedrals_with_mult = 0.20 # flat minimum force constant variation that fct_guess_fact shall yield, used to find low and high boundaries for random generation of particles' force constants
+bond_dist_guess_variation = 0.025  # nm
+angle_value_guess_variation = 10  # degrees
+dihedral_value_guess_variation = 10  # degrees
+fct_guess_min_flat_diff_bonds = 200  # flat minimum force constant variation that fct_guess_fact shall yield, used to find low and high boundaries for random generation of particles' force constants
+fct_guess_min_flat_diff_angles = 50  # flat minimum force constant variation that fct_guess_fact shall yield, used to find low and high boundaries for random generation of particles' force constants
+fct_guess_min_flat_diff_dihedrals_without_mult = 0.50  # flat minimum force constant variation that fct_guess_fact shall yield, used to find low and high boundaries for random generation of particles' force constants
+fct_guess_min_flat_diff_dihedrals_with_mult = 0.20  # flat minimum force constant variation that fct_guess_fact shall yield, used to find low and high boundaries for random generation of particles' force constants
 
 # gromacs functions that are properly treated at the moment
 # if we find a function that is not handled, program will exit with an appropriate error message
-# TODO: handle dihedral function 9 correctly so that different potentials can be stacked
-# TODO: for the same beads -- this is the primary purpose of function 9 !!
+# TODO: handle dihedral function 9 correctly so that different potentials can be stacked for the same beads
+#       this is the primary purpose of function 9 !!
 handled_functions = {
-    'contraint': [1],
-    'bond': [1],
-    'angle': [1, 2],
-    'dihedral': [1, 2, 4],
-    'dihedral_with_mult': [1, 4]
-    # these functions use 3 parameters, the last one being multiplicity (if it's omitted gromacs will use 1 by default, we reproduce this behavior)
+    'constraint': [1],  # tested and verified: 1
+    'bond': [1],  # tested and verified: 1
+    'angle': [1, 2],  # tested and verified: 1, 2
+    'dihedral': [1, 2, 4],  # tested and verified: 1, 2, 4 -- ongoing: 9 (need to merge the 1+ dihedrals groups on plots)
+    'virtual_sites2': [1],  # tested and verified: 1 -- ongoing: 2 (need GMX 2020)
+    'virtual_sites3': [1, 2],  # tested and verified: 1, 2 -- ongoing: 3, 4
```

Comment:
So here I let only the verified ones and the program checks that the ITP uses functions we have validated as working. For evolving the code you thus need to put into these arrays the functions you want to work on, if they are not there already.